src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -295,54 +383,141 @@ TaskSketcherElements::TaskSketcherElements(ViewProviderSketch *sketchView)
 
-    ui->autoSwitchBox->setChecked(hGrp->GetBool("Auto-switch to edge", true));
-    ui->namingBox->setChecked(hGrp->GetBool("Extended Naming", false));
+    // make filter items checkable
+    ui->listMultiFilter->blockSignals(true);
+    for (int i = 0; i < ui->listMultiFilter->count(); i++) {
+        QListWidgetItem* item = ui->listMultiFilter->item(i);
 
-    ui->comboBoxElementFilter->setEnabled(!isautoSwitchBoxChecked);
-    ui->comboBoxModeFilter->setEnabled(true);
+        item->setFlags(item->flags() | Qt::ItemIsUserCheckable);
 
-    slotElementsChanged();
+        item->setCheckState(Qt::Checked);
+    }
+    ui->listMultiFilter->setVisible(false);
+    ui->listMultiFilter->blockSignals(false);
+
+    this->installEventFilter(this);
+    ui->filterBox->installEventFilter(this);
+    ui->listMultiFilter->installEventFilter(this);
 }
 
 TaskSketcherElements::~TaskSketcherElements()
 {
-    try {
-        ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Elements");
-        hGrp->SetBool("Auto-switch to edge", ui->autoSwitchBox->isChecked());
-        hGrp->SetBool("Extended Naming", ui->namingBox->isChecked());
+    connectionElementsChanged.disconnect();
+}
+
+/* filter functions --------------------------------------------------- */
+
+void TaskSketcherElements::on_filterBox_stateChanged(int val){ 
+    if(ui->filterBox->checkState() == Qt::Checked)
+        ui->listMultiFilter->show();
+    else
+        ui->listMultiFilter->hide();
+
+    slotElementsChanged();
+}
+
+bool TaskSketcherElements::eventFilter(QObject* obj, QEvent* event)
+{
+    if (obj == (QObject*)ui->filterBox && event->type() == QEvent::Enter && ui->filterBox->checkState() == Qt::Checked) {
+        ui->listMultiFilter->show();
     }
-    catch (const Base::Exception&) {
+    else if (obj == (QObject*)ui->listMultiFilter && event->type() == QEvent::Leave) {
+        ui->listMultiFilter->hide();
     }
+    else if (obj == this && event->type() == QEvent::Leave) {
+        ui->listMultiFilter->hide();
+    }
+    return QWidget::eventFilter(obj, event);
+}
 
-    connectionElementsChanged.disconnect();
+void TaskSketcherElements::on_listMultiFilter_itemChanged(QListWidgetItem* item)
+{
+    ui->listMultiFilter->blockSignals(true);
+
+    if (item == ui->listMultiFilter->item(3)) { //3 is 'All geos'
+        for (int i = 4; i < ui->listMultiFilter->count(); i++) {
+            ui->listMultiFilter->item(i)->setCheckState(item->checkState());
+        }
+    }
+
+    ui->listMultiFilter->blockSignals(false);
+
+    updateVisibility();
+}
+
+void TaskSketcherElements::setItemVisibility(QListWidgetItem* item)
+{
+    /* index
+    0 => Normal
+    1 => Construction
+    2 => External
+    3 => all geos
+    4 => Point
+    5 => Line
+    6 => Circle
+    7 => Ellipse
+    8 => Arc
+    9 => Arc of Ellipse
+    10 => Hyperbola
+    11 => Parabola
+    12 => bspline
+    */

```

Comment:
Not very maintainable. Do same thing as `FilterValue` in constraints ?

Reply:
Same, it's clear and simple. Implementing a complex structure as in constraint is unecessary complexity. It's 20 times easier to understand than the filter value in constraints so I don't see how it would be less maintainable.